File: lib/rin/io/events/CursorButtonEvent.cpp
#include "rin/io/events/CursorButtonEvent.h"

namespace rin::io
{
    CursorButtonEvent::CursorButtonEvent(Window* inWindow, const CursorButton& inButton, const InputState& inState, const Vec2<>& inPosition,
        const Flags<InputModifier>& inModifiers) : Event(inWindow), button(inButton), state(inState), position(inPosition), modifiers(inModifiers)
    {
    }
}

File: Rin.Engine/Graphics/Windows/Events/CursorButtonEvent.cs
namespace Rin.Engine.Graphics.Windows.Events;

public class CursorButtonEvent : CursorEvent
{
    public required CursorButton Button;
    public required InputModifier Modifiers;
    public required InputState State;
    public bool IsAltDown => Modifiers.HasFlag(InputModifier.Alt);
    public bool IsControlDown => Modifiers.HasFlag(InputModifier.Control);
    public bool IsShiftDown => Modifiers.HasFlag(InputModifier.Shift);
}